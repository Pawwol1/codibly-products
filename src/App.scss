* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  background: #eddcd9;
  font-size: 20px;
  font-family: 'Courier New', Courier, monospace;
}

.App {
  height: 100vh;
  width: 100vw;
  display: flex;
  flex-direction: column;
  justify-content: stretch;
  align-items: center;
}
