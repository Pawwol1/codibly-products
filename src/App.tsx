import React from 'react';
import { Route, Routes } from 'react-router-dom';
import Main from './components/Main';
import ProductModal from './components/ProductModal/ProductModal';
import PageNotFound from './components/PageNotFound/PageNotFound';
import './App.scss';

const App = () => {
  return (
    <Routes>
      <Route path="/" element={<Main />} />
      <Route path="/page/:pageNumber" element={<Main />} />
      <Route path="/filter/:filterID" element={<Main />} />
      <Route path="/product/:productID" element={<ProductModal />} />
      <Route path="*" element={<PageNotFound />} />
    </Routes>
  );
};

export default App;
