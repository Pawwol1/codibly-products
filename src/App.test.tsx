import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from './App';
import ProductsContext from './context/products-context';
import userEvent from '@testing-library/user-event';

it('snapshot for App component', () => {
  const renderedComponent = renderer
    .create(
      <ProductsContext.Provider
        value={{
          products: [{ id: 1, name: 'cerulean', year: 2000, color: '' }],
          totalProducts: 12,
          page: 1,
          setPage: () => {},
          totalPages: 3,
          searchBoolean: false,
          setSearchBoolean: () => {},
          filteredProduct: [],
          setFilteredProduct: () => {},
          isLoading: false,
          setIsLoading: () => {},
        }}
      >
        <MemoryRouter initialEntries={['/']}>
          <App />
        </MemoryRouter>
      </ProductsContext.Provider>
    )
    .toJSON();
  expect(renderedComponent).toMatchSnapshot();
});

test('modal should open after a row click', async () => {
  render(
    <ProductsContext.Provider
      value={{
        products: [{ id: 1, name: 'cerulean', year: 2000, color: '' }],
        totalProducts: 12,
        page: 1,
        setPage: () => {},
        totalPages: 3,
        searchBoolean: false,
        setSearchBoolean: () => {},
        filteredProduct: [],
        setFilteredProduct: () => {},
        isLoading: false,
        setIsLoading: () => {},
      }}
    >
      <App />
    </ProductsContext.Provider>,
    { wrapper: BrowserRouter }
  );
  const user = userEvent.setup();

  const openModalElement = screen.getByTestId('productRow_testid');
  await user.click(openModalElement);

  const btnElement = screen.getByText(/go back/i);
  expect(btnElement).toBeInTheDocument();
});

test('should show "Page not Found" on badRoute', () => {
  const badRoute = '/bad/route';
  render(
    <MemoryRouter initialEntries={[badRoute]}>
      <App />
    </MemoryRouter>
  );
  expect(screen.getByText(/Page not Found/i)).toBeInTheDocument();
});
