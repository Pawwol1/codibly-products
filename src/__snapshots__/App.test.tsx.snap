// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for App component 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <h1>
      ReqRes Products
    </h1>
    <p>
      Application created with free ReqRes API
    </p>
  </header>
  <div
    className="filterInput"
  >
    <input
      onChange={[Function]}
      placeholder="Type ID to search"
      type="text"
      value=""
    />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInherit Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInherit css-1sim9df-MuiButtonBase-root-MuiButton-root"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      Search
      <span
        className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27c1.2-1.4 1.82-3.31 1.48-5.34-.47-2.78-2.79-5-5.59-5.34-4.23-.52-7.79 3.04-7.27 7.27.34 2.8 2.56 5.12 5.34 5.59 2.03.34 3.94-.28 5.34-1.48l.27.28v.79l4.25 4.25c.41.41 1.08.41 1.49 0 .41-.41.41-1.08 0-1.49L15.5 14zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </span>
    </button>
  </div>
  <table
    className="tableOfProducts"
  >
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Name
        </th>
        <th>
          Year
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="productRow"
        data-testid="productRow_testid"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td>
          1
        </td>
        <td>
          cerulean
        </td>
        <td>
          2000
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="paginationComponent"
  >
    <nav
      aria-label="pagination navigation"
      className="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
      data-testid="pagination"
    >
      <ul
        className="MuiPagination-ul css-wjh20t-MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateBeforeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-current="true"
            aria-label="page 1"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-selected MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            1
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 2"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            2
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 3"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            3
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
